### PR TITLE
Ftr/docker

### PR DESCRIPTION
**Pull request by SimonR99**
Fix zedcam dependency and add hability to run markhor (Gazebo, lidar, slam and nav) in a docker container.

To disable the camera you need an env variable

export MARKHOR_SIMULATION=true

To run the docker container : 
./scripts/start_gazebo.sh
docker-compose up

### CODE REVIEW

File: Dockerfile.base [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+# Install dependencies
+RUN rosdep install --from-paths src --ignore-src -r -y
+
+# Install the ROS bridge (for web control)
+RUN apt install -y ros-melodic-rosbridge-server
+RUN apt-get -y install rviz
+
+# Install the Velodyne simulator
+RUN apt install -y ros-melodic-velodyne-simulator
```
Comment (saxtot) [inline Dockerfile.base]: Should be in a package.xml.

File: README.md [hunk 2]
```diff
@@ -35,6 +35,14 @@ There are multiple launch configurations depending on your needs. The intended u
 - `markhor_base.launch` : While the two other files are made to be an optimised for competition setup, the `base` version launches
 everything simultaneously for demonstration purposes.  
 
+## Environnment variable
+
+You can add two environnments variables in your bashrc.
+
+GAZEBO_GUI : Enable/disable the GUI version of gazebo. If you don't want the gazebo ui, you can set it the false. The default value is true
+
+MARKHOR_SIMULATION : If set to true, it disable the use of the ZED ros wrapper. This way, you don't need to clone/install the ros wrapper/sdk. The default value is false.
```
Comment (saxtot) [inline README.md]: These variables are for Docker, no? As such, they should be in the corresponding section.

File: README.md
Comment (saxtot) [inline README.md:1]: Multiple typos, please run an auto corrector.

File: .github/workflows/docker-image.yml [hunk 3]
```diff
@@ -0,0 +1,64 @@
+name: Docker Image CI
+
+on:
+  push:
+    branches: [ "master", "enrich", "ftr/docker" ]
+  pull_request:
+    branches: [ "master", "enrich", "ftr/docker" ]
```
Comment (Leuchak) [inline .github/workflows/docker-image.yml:7]:
```suggestion
    branches: [ "*" ]
  pull_request:
    branches: [ "*" ]
```

**Review by saxtot — CHANGES_REQUESTED** [review]

**Review by saxtot — APPROVED** [review]

**Comment by saxtot** [comment]
This PR is already so long. Consider branching out for any other changes.

**Comment by Leuchak** [comment]
Don't wait for the pull request to become too large to be reviewed. It will end up like the 'moveit_minimal' branch in 'ovis,' remaining there forever without ever being merged. The addition of Docker is a feature, and all the components required for it to work are extras that should be documented in their own pull request. For example, 'add network connectivity' is not related to the addition of Docker.

File: docker-compose.yml [hunk 4]
```diff
@@ -4,16 +4,58 @@ version: '3.9'
 # Gazebo or the real hardware must be started separately
 # Gazebo can be started with ./scripts/start_gazebo.sh
 services:
+  markhor_gazebo:
```
Comment (saxtot) [inline docker-compose.yml:5]: So is Gazebo always launched by the compose file?

Comment (SimonR99) [reply docker-compose.yml:5]: Yes, but you can start only gazebo if you want. I'll add some doc about it, but you can do :

Start only gazebo : docker compose start markhor_gazebo
Start gazebo and slam : docker compose start markhor_gazebo markhor_slam

That way, you don't need a long command or some wizard script. For now, this is only supported on windows with x server, but I'll add support for linux in another pr as suggested by you and sam.

**Review by Leuchak — APPROVED** [review]
look good

**Comment by Leuchak** [comment]
my bad, It is --depth 1 and not --depth 0
> Error: buildx failed with: ERROR: failed to solve: executor failed running [/bin/sh -c cd ./src &&     git clone https://github.com/clubcapra/capra_estop.git &&     git clone --depth 0 https://github.com/lprobsth/gazebo_ros_tracked_vehicle_interface.git]: exit code: 128

**Pull request merged**